Formula/openttd.rb

```diff
@@ -39,18 +52,29 @@ class Openttd < Formula
   end
 
   def install
-    mkdir "build" do
-      system "cmake", "..", *std_cmake_args
-      system "make"
+    args = std_cmake_args
+    args << "-DCMAKE_INSTALL_BINDIR=bin" unless OS.mac?
+    system "cmake", "-G", "Ninja", "-S", ".", "-B", "build", *args
+    system "ninja", "-C", "build"
+    if OS.mac?
```

Comment:
Why a different build system for macOS?

Reply:
I'm wondering now if it was not so wise to try to switch to `ninja`, given the apparent failure.  This should also work with `cmake --build` and `cmake --install` even on Linux.  I'm going to test locally and make sure everything actually works this time, on both platforms.

Reply:
As is, the `cmake --install ~~> make install` doesn't seem to work properly on macOS. 

If we are okay without building `.app` bundle, we could try doing something similar to Linux. However, I noticed that the game had trouble finding the installed data unless I `cd`'d to corresponding directory like:
```console
$ cd /opt/homebrew/opt/openttd/share/games/openttd
$ openttd
```

Example (I also needed to install baseset into different directory to automatically pick up):
```rb
    # Disable CMake fixup_bundle to prevent copying dylibs
    inreplace "cmake/PackageBundle.cmake", "fixup_bundle(", "# \\0"

    system "cmake", "-S", ".", "-B", "build", *std_cmake_args,
                    "-DOPTION_INSTALL_FHS=ON",
                    "-DCMAKE_INSTALL_BINDIR=bin"
    system "cmake", "--build", "build"
    system "cmake", "--install", "build"

    resources.each do |r|
      (share/"games/openttd/baseset"/r.name).install r
    end
```

The problem is figuring out startup behavior to make sure command can be run from arbitrary directory. Also would need to check if any platform-specific logic for macOS will break.

Reply:
As is, the `cmake --install ~~> make install` doesn't seem to work properly on macOS. 

If we are okay without building `.app` bundle, we could try doing something similar to Linux. However, I noticed that the game had trouble finding the installed data unless I `cd`'d to corresponding directory like:
```console
$ cd /opt/homebrew/opt/openttd/share/games/openttd
$ openttd
```

Example (I also needed to install baseset into different directory to automatically pick up):
```rb
    # Disable CMake fixup_bundle to prevent copying dylibs
    inreplace "cmake/PackageBundle.cmake", "fixup_bundle(", "# \\0"

    system "cmake", "-S", ".", "-B", "build", *std_cmake_args,
                    "-DOPTION_INSTALL_FHS=ON",
                    "-DCMAKE_INSTALL_BINDIR=bin"
    system "cmake", "--build", "build"
    system "cmake", "--install", "build"

    resources.each do |r|
      (share/"games/openttd/baseset"/r.name).install r
    end
```

The problem is figuring out startup behavior to make sure command can be run from arbitrary directory. Also would need to check if any platform-specific logic for macOS will break.